impl/src/main/java/org/ehcache/config/builders/CacheConfigurationBuilder.java

```diff
@@ -216,13 +267,9 @@ private CacheConfigurationBuilder(CacheConfigurationBuilder<K, V> other) {
    * @param <T> the type of the service configuration
    * @return a matching service configuration, or {@code null} if none can be found
    */
-  public <T extends ServiceConfiguration<?>> T getExistingServiceConfiguration(Class<T> clazz) {
-    for (ServiceConfiguration<?> serviceConfiguration : serviceConfigurations) {
-      if (clazz.equals(serviceConfiguration.getClass())) {
-        return clazz.cast(serviceConfiguration);
-      }
-    }
-    return null;
+  @Deprecated
+  public <T extends ServiceConfiguration<?, ?>> T getExistingServiceConfiguration(Class<T> clazz) {
+    return getServices(clazz).iterator().next();
```

Comment:
This will throw if there's not matching configuration -- the old implementation returned a `null` (as is documented).

Reply:
Fixed.